### DISCUSSION TITLE
Application of existing patches when repatching a package.

### DISCUSSION THREAD

**Discussion by pleunv** [pnpm/pnpm#5181] [Ideas]
I'm currently looking into making the switch from yarn v1 to pnpm. One aspect I use quite a bit is patching of deps through patch-package, so seeing that native patching support has been added recently was really great!

One thing I do miss from patch-package that appears to be absent in pnpm's solution is a way to apply already existing patches to a package when running `pnpm patch <pkg>`. In patch-package this is supported by default since the package source you edit is the one available in your `node_modules` directory, which already had any previous patches applied to it during install. Since pnpm installs packages in a temporary directory this is not the case here. Is this something pnpm could support?

What could be a workaround for now is a way to manually apply patch files to the temporary package directory, but I can't immediately figure out a way to do that either.

**Comment by Lutymane** [comment]
<strike>+1

Regarding the work around, the files generated are git patch files, so `git apply package.patch` should work, but I guess some paths editing will be needed(?) It's more convenient to extract the package into some directory with `--edit-dir` and then only perform `patch-commit`s to maintain all the changes</strike>

**Comment by pleunv** [comment]
If I'm not mistaken this was implemented recently in [v7.25.0](https://github.com/pnpm/pnpm/releases/tag/v7.25.0). Is that not working for you @Lutymane?

**Reply by Lutymane** [reply]
> implemented recently in v7.25.0

Oh, somehow I missed my version was outdated... Thank you! It works now

**Comment by shovel-kun** [comment]
I'm still a little confused as to how to patch from a pre-existing patch file.

Say my co-worker gives me a .patch file. Is there a way to tell pnpm to automatically apply it to X package? 

I don't need/want to edit the files in `/node_modules` since I already have the changes.

**Comment by reslear** [comment]
prepare:
- use native unix patch or install `npm i -g patch-package`

```bash
// 1. go to new path 
cd node_modules/.pnpm_patches/@ckeditor/strapi-plugin-ckeditor@1.0.2

// 2.  (check is relative path works like: cd ../../../../patches) and run (unix native path/patch-package): 
patch -p1 -i ../../../../patches/@ckeditor__strapi-plugin-ckeditor.patch
```
